subprojects/jacoco/src/main/java/org/gradle/testing/jacoco/tasks/JacocoAggregatedReport.java

```diff
@@ -0,0 +1,166 @@
+import org.gradle.api.tasks.CacheableTask;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.SourceSet;
+import org.gradle.api.tasks.TaskContainer;
+import org.gradle.internal.jacoco.AntJacocoReport;
+import org.gradle.testing.jacoco.plugins.JacocoTaskExtension;
+
+import javax.inject.Inject;
+import java.io.File;
+import java.util.Collections;
+
+/**
+ * Task to aggregate HTML, XML and CSV reports of Jacoco coverage data from multiple projects and/or multiple Test tasks.
+ *
+ * @since 7.2
+ */
+@Incubating
+@CacheableTask
+public abstract class JacocoAggregatedReport extends JacocoReport {
+
+    public static final String AGGREGATION_CONFIGURATION_NAME = "jacocoAggregation";
+
+    /**
+     * Configures the test categories to be aggregated by this task.
+     * The tests category is the name of the test task.
+     *
+     * Defaults to ["test"].
+     */
+    @Input
+    public abstract ListProperty<String> getTestCategories();
+
+    @Inject
+    public JacocoAggregatedReport(JvmEcosystemUtilities jvmEcosystemUtilities) {
+        getTestCategories().convention(Collections.singletonList("test"));
+
+        Project project = getProject();
+        project.getPluginManager().withPlugin("java", plugin -> {
+            JavaPluginExtension javaPluginExtension = project.getExtensions().getByType(JavaPluginExtension.class);
+            // TODO: what about other "production" source sets that the users might add?
+            sourceSets(javaPluginExtension.getSourceSets().getByName(SourceSet.MAIN_SOURCE_SET_NAME));
+
+            executionData(getTestCategories().map(categories -> {
+                ConfigurableFileCollection coverageFiles = project.files();
+                for (String testCategory : categories) {
+                    TaskContainer tasks = project.getTasks();
+                    if (tasks.findByName(testCategory) != null) {
+                        coverageFiles.from(project.getTasks().named(testCategory).map(task -> task.getExtensions().getByType(JacocoTaskExtension.class).getDestinationFile()));
+                    }
+                }
+                return coverageFiles;
+            }));
+        });
+
+        Configuration aggregationConfiguration = project.getConfigurations().findByName(AGGREGATION_CONFIGURATION_NAME);
+        resolveClassesVariantsFrom(aggregationConfiguration, jvmEcosystemUtilities);
+        resolveSourcesVariantsFrom(aggregationConfiguration);
+        resolveTestCoverageDataVariantsFrom(aggregationConfiguration);
+    }
+
+    private void resolveClassesVariantsFrom(Configuration aggregationConfiguration, JvmEcosystemUtilities jvmEcosystemUtilities) {
+        Configuration coverageClassesDirs = createResolver("coverageClassesDirs");
+        coverageClassesDirs.extendsFrom(aggregationConfiguration);
+        coverageClassesDirs.getResolutionStrategy().getComponentSelection().all(s -> s.reject("external dependencies are excluded from code coverage report"));
```

Comment:
This is what you asked about a couple days ago, a way to ignore non project dependencies, right?

Reply:
That's right!

Reply:
This https://github.com/gradle/gradle/issues/17749 is very related.
Our current sample does not address this very common use case - project code coverage gets lost in the external dependencies that are included in the report.

Reply:
Could we do something like this? 
https://github.com/gradle/gradle/blob/5bdd1d6a740f6bf30675369fa61df7ef3aaf7f1b/subprojects/scala/src/main/java/org/gradle/api/plugins/scala/ScalaBasePlugin.java#L235-L241

This uses an artifact view with a component filter

Reply:
Thanks! This works nicely.